src/citrine/_serialization/properties.py

```diff
@@ -101,40 +135,54 @@ def deserialize_from_dict(self, data: dict) -> DeserializedType:
                 break
             else:
                 value = next_value
-        base_class = _get_base_class(data, self.serialization_path)
-        return self.deserialize(value, base_class=base_class)
+        return self.deserialize(value, base_class=None)  # always a dict
 
     def serialize_to_dict(self, data: dict, value: DeserializedType) -> dict:
         if self.serialization_path is None:
             raise ValueError('No serialization path set!')
-        else:
-            base_class = _get_base_class(data, self.serialization_path)
-            _data = data
-            fields = self.serialization_path.split('.')
-            for field in fields[:-1]:
-                _data = _data.setdefault(field, {})
-            _data[fields[-1]] = self.serialize(value, base_class=base_class)
-            return data
+
+        _data = data
+        fields = self.serialization_path.split('.')
+        for field in fields[:-1]:
+            _data = _data.setdefault(field, {})
+        _data[fields[-1]] = self.serialize(value, base_class=None)  # Always a dict
+        return data
 
     def __get__(self, obj, objtype=None) -> DeserializedType:
         """Property getter, deferring to the getter of the parent class, if applicable."""
-        base_class = _get_base_class(obj, self.serialization_path)
-        if base_class is not None and self.override:
-            return getattr(base_class, self.serialization_path).fget(obj)
+        if self._key is None:
+            property_name, base_class = _get_key_and_base_class(self, type(obj))
+            if base_class is not None:
+                return getattr(base_class, property_name).fget(obj)
+            else:
+                raise AttributeError(
+                    f"No parent class could be resolved for '{property_name}' "
+                    f"in {obj.__class__}.  'override' should probably be false."
+                )
         else:
             return getattr(obj, self._key, self.default)
 
     def __set__(self, obj, value: typing.Union[SerializedType, DeserializedType]):
         """Property setter, deferring to the setter of the parent class, if applicable."""
-        base_class = _get_base_class(obj, self.serialization_path)
+        property_name, base_class = _get_key_and_base_class(self, type(obj))
         if issubclass(type(value), self.underlying_types):
             value_to_set = value
         else:
             # if value is not an underlying type, set its deserialized version.
             value_to_set = self.deserialize(value, base_class=base_class)
 
-        if base_class is not None and self.override:
-            getattr(base_class, self.serialization_path).fset(obj, value_to_set)
```

Comment:
This was a latent bug.  `serialization_path` happened to equal all property_names for the cases where something was overridden, but that was definitely not guaranteed.